File: src/Piece.java
import java.util.List;

public abstract class Piece {
    public final Color color;
    public final int value;
    private final String whiteRepresentation;
    private final String blackRepresentation;

    public Piece(Color color, int value, String whiteRepresentation, String blackRepresentation) {
        this.color = color;
        this.value = value;
        this.whiteRepresentation = whiteRepresentation;
        this.blackRepresentation = blackRepresentation;
    }

    public abstract List<List<Event>> getValidMoves(Board board, Position position);

    public String repr() {
        return whiteRepresentation;
    }
}

File: src/Piece.kt
abstract class Piece(val color: Color, val value: Int, private val repr: String) {
    abstract fun getValidMoves(board: Board, position: Position, checkValidation: Boolean): List<Tempo>
    open fun couldBeCheck(board: Board, from: Position, to: Position) : Boolean {
        return true
    }
    open fun isCheckFast(board: Board, from: Position, to:Position) : Boolean {
        return false
    }
    fun repr(): String {
        return repr
    }
}